### PR TITLE
[Rest Client] LRO Helper Implementation

### PR DESCRIPTION
**Pull request by joheredi**
Implementing LRO Helper for Rest Clients

- [x] LRO Helper Implementation
- [x] LRO Test coverage
- [x] LRO Helper Generation

### CODE REVIEW

**Review by deyaaeldeen — APPROVED** [review]
🚀

File: src/restLevelClient/generateRestLevel.ts [hunk 1]
```diff
@@ -28,9 +30,20 @@ export async function generateRestLevelClient() {
     }
   });
 
+  model.operationGroups.some(og =>
+    og.operations.some(
+      o => o.extensions && o.extensions["x-ms-long-running-operation"]
+    )
+  );
```
Comment (qiaozha) [inline src/restLevelClient/generateRestLevel.ts:37]: Is this duplicate with what's in hasPollingOperations.ts ?

File: test/integration/generated/lroRest/src/lRORestClient.ts [hunk 2]
```diff
@@ -0,0 +1,1292 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (qiaozha) [inline test/integration/generated/lroRest/src/lRORestClient.ts:1]: Why is this filename seems a little weird ?

File: test/utils/test-swagger-gen.ts [hunk 3]
```diff
@@ -802,6 +802,17 @@ const testSwaggers: { [name: string]: SwaggerConfig } = {
     isTestPackage: true
   },
   // TEST REST LEVEL CLIENTS
+  lroRest: {
+    swaggerOrConfig: "lro.json",
+    clientName: "LRORestClient",
```
Comment (qiaozha) [inline test/utils/test-swagger-gen.ts:807]: It doesn't seem to be consistent with what's defined in here.

File: src/restLevelClient/generateRestLevel.ts (hunk 1 above)
Comment (joheredi) [reply src/restLevelClient/generateRestLevel.ts:37]: Yup, its a left over. Removing

File: test/integration/generated/lroRest/src/lRORestClient.ts (hunk 2 above)
Comment (joheredi) [reply test/integration/generated/lroRest/src/lRORestClient.ts:1]: I think Sarangan has a PR with some way to skip normalization. The naming logic tries to camel case, but it is not smart enough to identify acronyms.

File: src/restLevelClient/pollingHelper.ts.hbs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  response: TResult
+): LroResponse<TResult> {
+  if (Number.isNaN(response.status)) {
+    throw new TypeError(
+      `Status code of the response is not a number. Value: ${response.status}`
+    );
+  }
+
+  return {
+    flatResponse: response,
```
Comment (MaryGao) [inline src/restLevelClient/pollingHelper.ts.hbs:62]: I found the flatResponse is the whole response not logical result.

**Pull request merged**